rasa/shared/nlu/training_data/message.py

```diff
@@ -398,3 +398,18 @@ def is_e2e_message(self) -> bool:
             (self.get(ACTION_TEXT) and not self.get(ACTION_NAME))
             or (self.get(TEXT) and not self.get(INTENT))
         )
+
+    def find_overlapping_entities(
+        self,
+    ) -> List[Tuple[Dict[Text, Any], Dict[Text, Any]]]:
+        """Finds any overlapping entity annotations."""
+        entities = self.get("entities", [])[:]
+        entities.sort(key=lambda e: e["start"])
```

Comment:
I think some entities might not have a `start` key

Reply:
oh really, damn, do you know in which instances this might happen?

Reply:
I think mostly custom components: https://github.com/RasaHQ/rasa/issues/8114

Reply:
Okay, only using entities here for overlap search that have a start and end